### ISSUE TITLE
LossScaleOptimizer compatible with DecoupledWeightDecayExtension

### ISSUE THREAD

**Issue by ben-arnao** [tensorflow/tensorflow#38650]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.1
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**

The features allows for use decoupled weight decay optimizers

https://www.tensorflow.org/addons/api_docs/python/tfa/optimizers/weight_decay_optimizers/DecoupledWeightDecayExtension

with mixed precision using

https://www.tensorflow.org/api_docs/python/tf/keras/mixed_precision/experimental/LossScaleOptimizer

**Will this change the current api? How?**

Don't believe so

**Who will benefit with this feature?**

Those used decoupled weight decay and mixed precision.

**Any Other info.**

That's it.

**Comment by ravikyram** [comment]
@ben-arnao 

Do you have any use case that requires the feature you are interested in? Please feel free to submit a PR if you have use cases that supports that feature.Thanks!

**Comment by ben-arnao** [comment]
> @ben-arnao
> 
> Do you have any use case that requires the feature you are interested in? Please feel free to submit a PR if you have use cases that supports that feature.Thanks!

The use case would be doing mixed precision with weight decay decoupled optimizer and then using a callback like ReduceLrOnPlateua where we need access to modify weight decay dynamically. 

However it looks like this is not something TF wants to change at this time but maybe will be improved in future releases.

I know it's a narrow use case but for anyone that runs into the same issue, modifying the LossScaleOptimizer class to expose weight_decay works and you can decay LR and WD in the same manner.

#38657
